src/Processors/Executors/PullingAsyncPipelineExecutor.cpp

```diff
@@ -36,8 +37,11 @@ struct PullingAsyncPipelineExecutor::Data
 
 PullingAsyncPipelineExecutor::PullingAsyncPipelineExecutor(QueryPipeline & pipeline_) : pipeline(pipeline_)
 {
-    lazy_format = std::make_shared<LazyOutputFormat>(pipeline.getHeader());
-    pipeline.setOutput(lazy_format);
+    if (!pipeline.isCompleted())
```

Comment:
We need it, because in case of `insert select` we don't need to send any data to client (except progress).
In other words, out pipeline works without output format - it is fully completed and need to be executed as is.

Another one `if` will not change performance of course.
BTW, perftests for insert select in multiple threads are faster.